packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -421,22 +574,45 @@ impl CreateTransactionBuilder {
         }
     }
 
-    fn resolve_fuel_tx(self, base_offset: usize, num_witnesses: u8) -> Result<Create> {
-        let num_of_storage_slots = self.storage_slots.len();
+    async fn build_with_provider(self, _: &impl DryRunner) -> Result<CreateTransaction> {
+        self.build()
+    }
+
+    pub fn build(self) -> Result<CreateTransaction> {
+        let is_using_predicates = self.is_using_predicates();
+        let base_offset = if is_using_predicates {
+            self.base_offset()
+        } else {
+            0
+        };
 
-        let gas_price = self.gas_price.unwrap_or(self.network_info.min_gas_price);
-        let gas_limit = self.gas_limit.unwrap_or(self.network_info.max_gas_per_tx);
+        let num_witnesses = self.num_witnesses()?;
+        let tx = self.resolve_fuel_tx(base_offset, num_witnesses)?;
+
+        Ok(CreateTransaction {
+            tx,
+            is_using_predicates,
+        })
+    }
+
+    fn resolve_fuel_tx(self, mut base_offset: usize, num_witnesses: u8) -> Result<Create> {
+        let mut policies = self.generate_common_fuel_policies();
+        policies.set(
+            PolicyType::WitnessLimit,
+            self.witness_limit.or(Some(DEFAULT_CREATE_WITNESS_LIMIT)),
+        );
+
+        let storage_slots_offset = self.storage_slots.len() * StorageSlot::SLOT_SIZE;
+        base_offset += storage_slots_offset + policies.size_dynamic();
```

Comment:
This bit here could also use some _why_ comments.

Reply:
Nothing to comment. That is just how the `vm` expects the offset. base_offset + storage_slot_offset + whatever was set with policies.